src/Orleans.Transactions/State/ReaderWriterLock.cs

```diff
@@ -1,14 +1,57 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Numerics;
+using System.Threading;
 using System.Threading.Tasks;
+using System.Transactions;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Orleans.Configuration;
 using Orleans.Runtime;
 using Orleans.Transactions;
 using Orleans.Transactions.Abstractions;
 
 namespace Orleans.Transactions
 {
+    public class TransactionStateStatistics
```

Comment:
This file is already 500+ lines, could you put it in another file?

Reply:
yes.  I'm actually not sure to check this statistic class in since it has problems and is not the goal for this pr, as I mentioend in the description.  I will move it to another file if I decided to check it in.